pkg/gatewayserver/io/udp/udp.go

```diff
@@ -96,6 +108,16 @@ func (s *srv) read() (err error) {
 			}
 			return err
 		}
+
+		// TODO: Should UDP port be a part of this?
```

Comment:
We need to re-think the connection logic of UDP gateways anyway (https://github.com/TheThingsNetwork/lorawan-stack/issues/2492). Then we can rate limit _new_ "connections" by IP address, and rate limit messages on each individual "connection".